src/sentry/static/sentry/app/views/organizationGroupDetails/reprocessingDialogForm.tsx

```diff
@@ -53,46 +59,69 @@ function ReprocessingDialogForm({group, organization, Header, Body, closeModal}:
     browserHistory.push(
       `/organizations/${orgSlug}/issues/?query=reprocessing.original_issue_id:${group.id}/`
     );
-  }
+  };
+
+  handleError = () => {
+    addErrorMessage(t('Failed to reprocess. Please check your input.'));
+  };
 
-  return (
-    <React.Fragment>
-      <Header closeButton>{title}</Header>
-      <Body>
-        <Introduction>
-          {t(
-            'Reprocessing applies any new debug files or grouping configuration to an Issue. Before you give it a try, you should probably consider these impacts:'
-          )}
-        </Introduction>
-        <StyledList symbol="bullet">
-          {impacts.map((impact, index) => (
-            <ListItem key={index}>{impact}</ListItem>
-          ))}
-        </StyledList>
-        <ApiForm
-          apiEndpoint={endpoint}
-          apiMethod="POST"
-          footerClass="modal-footer"
-          onSubmitSuccess={handleSuccess}
-          submitLabel={title}
-          submitLoadingMessage={t('Reprocessing\u2026')}
-          submitErrorMessage={t('Failed to reprocess. Please check your input.')}
-          onCancel={closeModal}
-        >
-          <NumberField
-            name="maxEvents"
-            label={t('Enter the number of events to be reprocessed')}
-            help={tct(
-              'You can limit the number of events reprocessed in this Issue. If you set a limit, we will reprocess your most recent events, [strong:and the rest will be deleted.]',
-              {strong: <strong />}
+  render() {
+    const {group, organization, Header, Body, closeModal} = this.props;
+    const {maxEvents} = this.state;
+    const orgSlug = organization.slug;
+    const endpoint = `/organizations/${orgSlug}/issues/${group.id}/reprocessing/`;
+    const title = t('Reprocess Events');
+
+    return (
+      <React.Fragment>
+        <Header closeButton>{title}</Header>
+        <Body>
+          <Introduction>
+            {t(
+              'Reprocessing applies any new debug files or grouping configuration to an Issue. Before you give it a try, you should probably consider these impacts:'
             )}
-            placeholder={t('Reprocess all events')}
-            min={1}
-          />
-        </ApiForm>
-      </Body>
-    </React.Fragment>
-  );
+          </Introduction>
+          <StyledList symbol="bullet">
+            {impacts.map((impact, index) => (
+              <ListItem key={index}>{impact}</ListItem>
+            ))}
+          </StyledList>
+          <Form
+            submitLabel={title}
+            apiEndpoint={endpoint}
+            apiMethod="POST"
+            initialData={{maxEvents: '', remainingEvents: 'keep'}}
```

Comment:
I would create a const, like:

```
const remainingEventsOptions = [
                ['keep', t('Keep')],
                ['delete', t('Delete')],
              ]
```

and reuse it here. I like this approach because it makes it easy when you have to do some updating, so you only need to update in one place and you can also reuse it in tests 😉